kotlin/identify-raster-cell/src/main/java/com/esri/arcgisruntime/sample/identifyrastercell/MainActivity.kt

```diff
@@ -0,0 +1,142 @@
+      Raster(getExternalFilesDir(null)?.path + "/SA_EVI_8Day_03May20/SA_EVI_8Day_03May20.tif")
+
+    // create the layer
+    rasterLayer = RasterLayer(rasterFile)
+
+    // define a new map
+    val rasterMap = ArcGISMap(Basemap.Type.OCEANS, -33.9, 18.6, 9).apply {
+      // add the raster layer
+      operationalLayers.add(rasterLayer)
+    }
+
+    mapView.apply {
+      // add the map to the map view
+      map = rasterMap
+
+      // set behavior for double touch drag and on up gestures
+      onTouchListener = object : DefaultMapViewOnTouchListener(this@MainActivity, mapView) {
+        override fun onDoubleTouchDrag(e: MotionEvent): Boolean {
+          val offset = 200
+          // identify the pixel at a vertical offset from the motion event
+          identifyPixel(Point(e.x.toInt(), e.y.toInt() - offset))
+          return true
+        }
+
+        override fun onUp(e: MotionEvent): Boolean {
+          // identify the pixel at the given screen point
+          identifyPixel(Point(e.x.toInt(), e.y.toInt()))
+          return true
+        }
+      }
+    }
+  }
+
+  /**
+   * Identify the pixel at the given screen point and report raster cell attributes in a callout.
+   *
+   * @param screenPoint from motion event, for use in identify
+   */
+  private fun identifyPixel(screenPoint: Point) {
+    rasterLayer?.let { rasterLayer ->
+      // identify at the tapped screen point
+      val identifyResultFuture =
+        mapView.identifyLayerAsync(rasterLayer, screenPoint, 1.0, false, 10)
+
+      identifyResultFuture.addDoneListener {
+        // get the identify result
+        val identifyResult = identifyResultFuture.get()
+
+        // create a string builder
+        val stringBuilder = StringBuilder()
+
+        // get the a list of geoelements as raster cells from the identify result
+        identifyResult.elements.filterIsInstance<RasterCell>().forEach { cell ->
+          // get each attribute for the cell
+          cell.attributes.forEach {
+            // add the key/value pair to the string builder
+            stringBuilder.append(it.key + ": " + it.value)
+            stringBuilder.append("\n")
+          }
+
+          // format the X & Y coordinate values of the raster cell to a human readable string
+          val xyString =
+            "X: ${String.format("%.4f", cell.geometry.extent.xMin)} " +
+                "Y: ${String.format("%.4f", cell.geometry.extent.yMin)}"
+          // add the coordinate string to the string builder
+          stringBuilder.append(xyString)
+
+          // create a textview for the callout
+          val calloutContent = TextView(applicationContext).apply {
+            setTextColor(Color.BLACK)
+            // format coordinates to 4 decimal places and display lat long read out
+            text = stringBuilder.toString()
+          }
+          // display the callout in the map view
+          mapView.callout.apply {
+            location = mapView.screenToLocation(screenPoint)
```

Comment:
I was dragging my finger around and actually got an error at this line: `java.lang.IllegalArgumentException: Parameter location must not be null`. It crashed the app so there should probably be some kind of null check on the screen point.